pkg/liqo-controller-manager/resource-request-controller/offerUpdater.go

```diff
@@ -221,23 +202,60 @@ func (u *OfferUpdater) getStorageClasses(ctx context.Context) ([]sharingv1alpha1
 	return storageTypes, nil
 }
 
-// Push add new clusterID to update queue which will be processes as soon as possible.
-func (u *OfferUpdater) Push(cluster discoveryv1alpha1.ClusterIdentity) {
-	u.queue.Add(cluster)
+// SetThreshold sets the threshold for resource updates to trigger an update of the ResourceOffers.
+func (u *OfferUpdater) SetThreshold(updateThresholdPercentage uint) {
+	u.updateThresholdPercentage = updateThresholdPercentage
+	u.NotifyChange()
+}
+
+// isAboveThreshold checks if the resources have changed by at least updateThresholdPercentage since the last update.
+func (u *OfferUpdater) isAboveThreshold(clusterID string) bool {
+	oldResources := u.currentResources[clusterID]
+	newResources := u.ResourceReader.ReadResources(clusterID)
+	// Check for any resources removed
+	for oldResourceName := range oldResources {
+		if _, exists := newResources[oldResourceName]; !exists {
+			return true
+		}
+	}
+	// Check for any resources added
+	for newResourceName := range newResources {
+		if _, exists := oldResources[newResourceName]; !exists {
+			return true
+		}
+	}
+	for resourceName, newValue := range newResources {
+		oldValue := oldResources[resourceName]
+		absDiff := math.Abs(float64(newValue.Value() - oldValue.Value()))
+		if int64(absDiff) > oldValue.Value()*int64(u.updateThresholdPercentage)/100 {
+			return true
+		}
+	}
+
+	return false
 }
 
-// Remove removes a specified clusterID from the update queue and it will be no more processed.
-func (u *OfferUpdater) Remove(cluster discoveryv1alpha1.ClusterIdentity) {
-	u.queue.Forget(cluster)
-	klog.Infof("Removed cluster %s from update queue", cluster.ClusterName)
+// resourceIsEmpty checks if the ResourceList is empty.
+func resourceIsEmpty(list corev1.ResourceList) bool {
+	for _, val := range list {
+		if !val.IsZero() {
+			return false
+		}
+	}
+	return true
 }
 
-func getRandomTimeout() time.Duration {
-	max := new(big.Int)
-	max.SetInt64(int64(maxRandom))
-	n, err := rand.Int(rand.Reader, max)
+// GetResourceRequests returns the list of ResourceRequests for the given cluster.
+func GetResourceRequests(ctx context.Context, k8sClient client.Client, clusterID string) (
```

Comment:
Since you are looking for a single resourcerequest, you could include here the logic to check whether it exists or is duplicated, and return the appropriate error.